security/webauthn.go

```diff
@@ -18,28 +19,36 @@ type WebAuthNService struct {
 	domain       string
 }
 
-var AUTHORIZATION_COOKIE_KEY = "Authorization"
+const AUTHORIZATION_COOKIE_KEY = "Authorization"
 
-func NewWebAuthNService(userDB *userDB, origin string, domain string, jwtService *JwtService) *WebAuthNService {
-	var err error
+func NewWebAuthNService(userDB *userDB, origin string, domain string, jwtService *JwtService) (*WebAuthNService, error) {
 	webAuthn, err := webauthn.New(&webauthn.Config{
 		RPDisplayName: "equipment watchdog", // Display Name for your site
 		RPID:          domain,               // Generally the domain name for your site
 		RPOrigin:      origin,               // The origin URL for WebAuthn requests
 	})
 
 	if err != nil {
-		log.Fatal("Error creating webAuthn", err)
+		return nil, fmt.Errorf("could not create webAuth: %w", err)
 	}
 
 	sessionStore, err := session.NewStore()
 	if err != nil {
 		log.Fatal("Error creating sessionStore", err)
+		return nil, fmt.Errorf("could not create session-store: %w", err)
 	}
 
-	return &WebAuthNService{webAuthn: webAuthn, sessionStore: sessionStore, jwtService: jwtService, userDB: userDB, domain: domain}
+	return &WebAuthNService{
+		webAuthn:     webAuthn,
+		sessionStore: sessionStore,
+		jwtService:   jwtService,
+		userDB:       userDB,
+		domain:       domain,
+	}, nil
 }
 
+// FIXME: not really nice solution to cancle on this way the context - channel issue
```

Comment:
@am3o  What would be the correct way to handle errors? I do abort and return. `c.AbortWithError` cancels the request and therefore the following handlers, but still runs the whole current function. Therefore I use the early exit.
```
if err != nil {
  c.AbortWithError(http.StatusBadRequest, err)
  return
}
```
Is that the correct way?